### PR TITLE
O(1) solution

### PR DESCRIPTION
**Pull request by 6zz**
**!!! WARN !!! Do not merge it to master!!!**
could you please put this on a separate branch instead of Master?

### CODE REVIEW

File: 01-not-an-array.js [hunk 1]
```diff
@@ -26,19 +26,97 @@
...
-  unshift() {}
+    return (node) ? node.v : undefined;
+  }
+
+  // auxillaries
+  isEmpty() {
+      return this.head == null; // treat undefined as null
+  }
+
+  get storage() {
```
Comment (6zz) [inline 01-not-an-array.js]: is this cheating?  this getter function ensure access to this.storage conforms to the initial format: `{0: v1, 1: v2,...}`

**Comment by jasonly** [comment]
Bhahahah! SHAWWWWN! You did the question I had in mind for the next week hahaha! Looks good to me!

File: 01-not-an-array.js (hunk 1 above)
Comment (jasonly) [reply 01-not-an-array.js]: Hahahha nah!

**Comment by 6zz** [comment]
Oh! You mentioned O(1) in one of our conversations so I thought maybe the optimization is go achieve O(1) for all the four main methods of this class. :D  But I was not sure if it is legit by using getter function to manipulate the storage property.

**Pull request merged**